File: 0011-container-with-most-water/0011-container-with-most-water.php
class Solution {

  /**
    * @param Integer[] $height
    * @return Integer
    */
  function maxArea($height) {
    $ans = 0;
    $l = 0;
    $r = count($height) - 1;
    while ($l !== $r) {
      if ($height[$l] < $height[$r]) {
        $ans = max($ans, $height[$l] * ($r - $l));
        $l++;
      } else {
        $ans = max($ans, $height[$r] * ($r - $l));
        $r--;
      }
    }
    return $ans;
  }
}

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
  def maxArea(self, height: List[int]) -> int:
    ans, l, r = 0, 0, len(height) - 1
    while l != r:
      if height[l] < height[r]:
        ans = max(ans, height[l] * (r - l))
        l += 1
      else:
        ans = max(ans, height[r] * (r - l))
        r -= 1
    return ans
        